src/Task/TaskCollection.php

```diff
@@ -15,7 +15,7 @@
  */
 class TaskCollection extends Collection
 {
-    protected function throwNotFound(string $name)
+    protected function throwNotFound(string $name): void
```

Comment:
So, is this `: void` parts also necessary? 
We have lots and lots of anon function in recipes, do we really want to add it there? What benefits will it provide? 
Should we omit recipes/ all together?

Reply:
I would skip recipes, however IMO the core library should contain as much of native and phpDoc-based type definitions as possible.

As for void being hinted, IMO yes. This having explicit void means that function should never, ever, return anything. This in turn helps with finding bugs even on static analysis level, as PHPStan will report an issue if a piece of code tries to use a void result.